polygeist/test/polygeist-opt/sycl/sycl-raise-host.mlir

```diff
@@ -818,3 +818,223 @@ llvm.func @raise_set_captured(%handler: !llvm.ptr) {
+llvm.mlir.global private unnamed_addr constant @kernel_param_desc_str("kernel_param_desc\00")
+
+// CHECK-LABEL:   llvm.func @raise_schedule_parallel_for_range_accessor(
+// CHECK-SAME:                                                          %[[VAL_0:.*]]: !llvm.ptr) {
+// CHECK:           %[[VAL_1:.*]] = llvm.mlir.constant(0 : i32) : i32
+// CHECK:           %[[VAL_2:.*]] = llvm.mlir.constant(1 : i32) : i32
+// CHECK:           %[[VAL_3:.*]] = llvm.mlir.constant(2 : i32) : i32
+// CHECK:           %[[VAL_4:.*]] = llvm.mlir.null : !llvm.ptr
+// CHECK:           %[[VAL_5:.*]] = llvm.alloca %[[VAL_2]] x !llvm.struct<"class.sycl::_V1::range", opaque> : (i32) -> !llvm.ptr
+// CHECK:           sycl.host.handler.set_nd_range %[[VAL_0]] -> range %[[VAL_5]] : !llvm.ptr, !llvm.ptr
+// CHECK:           %[[VAL_6:.*]] = llvm.alloca %[[VAL_2]] x !llvm.struct<"class.lambda", opaque> : (i32) -> !llvm.ptr
+// CHECK:           %[[VAL_7:.*]] = llvm.alloca %[[VAL_2]] x !llvm.struct<"class.sycl::_V1::accessor", opaque> : (i32) -> !llvm.ptr
+// CHECK:           sycl.host.set_captured %[[VAL_6]][0] = %[[VAL_7]] : !llvm.ptr, !llvm.ptr (!sycl_accessor_1_21llvm2Evoid_rw_gb)
+// CHECK:           %[[VAL_8:.*]] = llvm.mlir.addressof @_ZN4sycl3_V16detailL17kernel_signaturesE : !llvm.ptr
+// CHECK:           %[[VAL_9:.*]] = llvm.mlir.addressof @kernel_num_params_str : !llvm.ptr
+// CHECK:           %[[VAL_10:.*]] = llvm.mlir.addressof @kernel_param_desc_str : !llvm.ptr
+// CHECK:           %[[VAL_11:.*]] = llvm.alloca %[[VAL_2]] x i32 : (i32) -> !llvm.ptr
+// CHECK:           %[[VAL_12:.*]] = llvm.alloca %[[VAL_2]] x !llvm.ptr : (i32) -> !llvm.ptr
+// CHECK:           llvm.store %[[VAL_3]], %[[VAL_11]] : i32, !llvm.ptr
+// CHECK:           llvm.store %[[VAL_8]], %[[VAL_12]] : !llvm.ptr, !llvm.ptr
+// CHECK:           "llvm.intr.var.annotation"(%[[VAL_11]], %[[VAL_9]], %[[VAL_4]], %[[VAL_1]], %[[VAL_4]]) : (!llvm.ptr, !llvm.ptr, !llvm.ptr, i32, !llvm.ptr) -> ()
+// CHECK:           "llvm.intr.var.annotation"(%[[VAL_12]], %[[VAL_10]], %[[VAL_4]], %[[VAL_1]], %[[VAL_4]]) : (!llvm.ptr, !llvm.ptr, !llvm.ptr, i32, !llvm.ptr) -> ()
+// CHECK:           sycl.host.schedule_kernel %[[VAL_0]] -> @device_functions::@foo[range %[[VAL_5]]](%[[VAL_7]]: !sycl_accessor_1_21llvm2Evoid_rw_gb) : (!llvm.ptr, !llvm.ptr, !llvm.ptr) -> ()
+// CHECK:           llvm.return
+// CHECK:         }
+llvm.func @raise_schedule_parallel_for_range_accessor(%handler: !llvm.ptr) {
+  %c0 = llvm.mlir.constant (0 : i32) : i32
+  %c1 = llvm.mlir.constant (1 : i32) : i32
+  %c2 = llvm.mlir.constant (2 : i32) : i32
+  %nullptr = llvm.mlir.null : !llvm.ptr
+
+  %range = llvm.alloca %c1 x !llvm.struct<"class.sycl::_V1::range", opaque> : (i32) -> !llvm.ptr
+  sycl.host.handler.set_nd_range %handler -> range %range : !llvm.ptr, !llvm.ptr
+
+  %lambda = llvm.alloca %c1 x !lambda_class : (i32) -> !llvm.ptr
+  %accessor = llvm.alloca %c1 x !llvm.struct<"class.sycl::_V1::accessor", opaque> : (i32) -> !llvm.ptr
```

Comment:
Can we add a `sycl.host.constructor` constructing the accessor here to check it all works fine?

Reply:
Done, though I believe that doesn't have an effect on the pattern because it just trusts the type annotated to the `set_captured` op.